### PR TITLE
1259 multi select fixes

### LINKED ISSUE #1259: MultiSelect Keyboard Navigation issues.
### Describe the bug

Hello
I have found 2 issues with the MultiSelect:
1 - MultiSelect is not usable with the keyboard, causing accessibility issues.
2 - When used in a form within a Modal, a MultiSelect will stop the focus event (via tab) from propagating past the MultiSelect, causing other inputs in the form to be unreachable via keyboard.

I was working on a fix for (1) as I thought both issues were related, but later discovered they are 2 distinct issues when (2) did not happen in my minimum reproduction until I added a modal. I suspect the modal's focustrap is involved for (2).

I'm attempting a fix on my side and may send a MR if I manage to find the problem, but I'm not the most experienced web developer so I may be out of my depth.

### Reproduction

``` javascript
<script>
    import { Label, Input, Helper, MultiSelect, Modal } from 'flowbite-svelte';
    
    let selected = [];
    let countries = [
        { value: 'us', name: 'United States' },
        { value: 'ca', name: 'Canada' },
        { value: 'fr', name: 'France' },
        { value: 'jp', name: 'Japan' },
        { value: 'en', name: 'England' }
    ];

</script>

<Modal open={true}>
    <form method="POST" enctype="multipart/form-data">
        <Label class="block mb-2">Username</Label>
        <Input label="Name" id="username" name="username" required/>

        <Label class="block mb-2">Favorite Countries</Label>
        <MultiSelect items={countries} bind:value={selected}/>
        <Helper class="text-sm mt-2">
            You can't access this field with the keyboard, unless there is data in it. And when you do, you can only remove the data, not add more.
            Also, if in a modal, this blocks focus to the next input, making the form unusable with keyboard.
        </Helper>
        <Label class="block mb-2">Your email</Label>
        <Input label="Email" id="email" name="email" required placeholder="name@flowbite.com" />
        <Helper class="text-sm mt-2">
            The Modal blocks tab navigation to this input. You can't access it with the keyboard (unless going backwards with shift+tab).
        </Helper>
    </form>
</Modal>
```

Using the code above, notice how it's impossible to add elements to the multiselect with the keyboard. Also notice how pressing tab from the Username input does not switch the focus to the Email input.

### Flowbite version and System Info

```shell
System:
    OS: Windows 11 10.0.22621
    CPU: (24) x64 12th Gen Intel(R) Core(TM) i9-12900
    Memory: 33.42 GB / 63.73 GB
  Binaries:
    Node: 18.15.0 - ~\AppData\Roaming\JetBrains\WebStorm2023.3\node\node-v18.15.0-win-x64\node.EXE
    npm: 9.5.0 - ~\AppData\Roaming\JetBrains\WebStorm2023.3\node\node-v18.15.0-win-x64\npm.CMD
  Browsers:
    Edge: Chromium (122.0.2365.52)
    Internet Explorer: 11.0.22621.1
(Chrome is not listed here somehow, but I use Chrome 121.0.6167.185 (Official Build) (64-bit))
  npmPackages:
    @sveltejs/kit: ^2.0.0 => 2.4.3
    flowbite-svelte: ^0.44.22 => 0.44.22
    svelte: ^4.2.7 => 4.2.8
    vite: ^5.0.3 => 5.0.12
```

### PR DESCRIPTION
**Pull request by herronelou**
Closes #1259 

## Status

- [ ] Not Completed
- [x] Completed

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] I have checked the page with https://validator.unl.edu/
- [x] All the tests have passed
- [x] My pull request is based on the latest commit (not the npm version).

## Summary by CodeRabbit

- **New Features**
    - Enhanced the MultiSelect component with keyboard navigation. Users can now use arrow keys to navigate, Enter or Space to select, and Escape to close the dropdown.
    - Improved visibility and interaction of the MultiSelect component with new styling for the active item.
- **Bug Fixes**
    - Adjusted the focus trap utility to correctly skip elements that are hidden, ensuring better accessibility and user experience.

### CODE REVIEW

**Comment by herronelou** [comment]
@shinokada are you the one who reviews PRs?

**Comment by shinokada** [comment]
Is it possible to create an example that you solved?
Your example and explanation can be at https://flowbite-svelte.com/docs/forms/select#MultiSelect .

**Comment by herronelou** [comment]
Hello.
What kind of examples are you looking for?
I have posted an example in the linked issue #1259

**Comment by shinokada** [comment]
Yes, that will be fine. Can you add it to the docs?

**Comment by herronelou** [comment]
I can, I'm just not too sure what to add. Every other component tends to
have keyboard navigation by default based on the original HTML elements,
this is just trying to mimic it.

Should I add a short mention of the keys that have been implemented?

**Comment by herronelou** [comment]
@shinokada I've added a little section in the docs, I'm not sure if this is what you had in mind?

**Comment by shinokada** [comment]
You wrote, "then press enter or spacebar to toggle the highlighted item.". But enter or spacebar don't work to select items.

**Comment by herronelou** [comment]
Hi @shinokada, it works for me here, can you give me more information about what doesn't work for you? 
Upon highlighting an element with the up/down arrows, pressing space or enter should toggle that element.

**Comment by shinokada** [comment]
Go to https://flowbite-svelte-lca18iqht-themesberg.vercel.app/docs/forms/select#Basic_example and use arrow down/up and hit the return key. It doesn't select an item. All MultiSelect examples do not work with return or space key. 

I tested with Chrome and FF.

**Comment by herronelou** [comment]
It doesn't work for me either on the link above, nor on my local build in the docs, but it works with my local build in my real project. I'll try to see if I can make it work in the docs context.

```
  System:
    OS: Windows 11 10.0.22631
    CPU: (24) x64 12th Gen Intel(R) Core(TM) i9-12900
    Memory: 31.99 GB / 63.73 GB
  Binaries:
    Node: 18.15.0 - ~\AppData\Roaming\JetBrains\WebStorm2023.3\node\node-v18.15.0-win-x64\node.EXE
    npm: 9.5.0 - ~\AppData\Roaming\JetBrains\WebStorm2023.3\node\node-v18.15.0-win-x64\npm.CMD
    pnpm: 8.15.6 - ~\AppData\Roaming\JetBrains\WebStorm2023.3\node\node-v18.15.0-win-x64\pnpm.CMD
  Browsers:
    Edge: Chromium (123.0.2420.97)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    @sveltejs/kit: ^2.5.1 => 2.5.1
    svelte: ^4.2.11 => 4.2.11
    vite: ^5.1.4 => 5.1.4
```

File: src/lib/forms/MultiSelect.svelte [hunk 1]
```diff
@@ -66,6 +71,40 @@
...
+      if (show) {
+        event.stopPropagation();  // We're handling the escape key, so prevent the default behavior, for example, closing a modal
+        show = false;
+      }
+    }
+    else if (event.key === `Enter` || event.key === ` `) {
+      // If the focused element is our div, prevent the form submission and show the dropdown
+      if (focusedElement?.role === 'listbox') {
+        event.preventDefault(); // prevent enter key from triggering form submission
+        // Show the dropdown when pressing enter/space while in focus.
```
Comment (herronelou) [inline src/lib/forms/MultiSelect.svelte]: @shinokada This check is preventing it from working in the context of the documentation, as the focusedElement is 
```
<iframe class="w-full h-full" title="iframe-code-content"></iframe>
```
I don't think this was strictly necessary but I was trying to be conservative.
I can remove the check, which would simplify the function, and make it work in the docs.
Again I'm normally a python dev so all this is quite new to me, I'd love your advice on the matter.

**Comment by herronelou** [comment]
@shinokada I have rebased this on the latest main branch, also cleaned up the commits into 3 clearly defined commits, while taking into account the comments from code rabbit.

Let me know if anything else is needed.

**Comment by shinokada** [comment]
Thank you for your contribution.

**Pull request merged**